weblate/addons/models.py

```diff
@@ -42,24 +42,42 @@
 
 
 class AddonQuerySet(models.QuerySet):
+    def filter_for_execution(self, component=None, project=None):
+        queryset = self.filter_sitewide()
+        # Include by component or project
+        if component:
+            queryset |= self.filter_component(component)
+        if project:
+            queryset |= self.filter_project(project)
+        return queryset.distinct()
+
     def filter_component(self, component):
         return self.prefetch_related("event_set").filter(
-            (Q(component=component) & Q(project_scope=False))
-            | (Q(component__project=component.project) & Q(project_scope=True))
-            | (Q(component__linked_component=component) & Q(repo_scope=True))
-            | (Q(component=component.linked_component) & Q(repo_scope=True))
+            Q(component=component)
+            | Q(project=component.project)
+            | Q(component__project=component.project)
+        )
+
+    def filter_project(self, project):
+        return self.prefetch_related("event_set").filter(project=project)
+
+    def filter_sitewide(self):
+        return self.prefetch_related("event_set").filter(
+            Q(component__isnull=True, project__isnull=True)
```

Comment:
Q is not needed here:

```suggestion
            component__isnull=True, project__isnull=True
```